torch/_dynamo/variables/nn_module.py

```diff
@@ -82,6 +82,19 @@ def unpack_var_sequence(self, tx):
         # implement list/iter/tuple/etc calls
         base = tx.output.get_submodule(self.module_key)
         options = VariableTracker.propagate([self])
+        if isinstance(base, torch.nn.ModuleDict):
+            result = {}
+            for name, submod in base.items():
+                name_var = variables.ConstantVariable(name)
+                result[name_var] = tx.output.register_attr_or_module(
+                    submod,
+                    self.module_key,
+                    name,
+                    source=NNModuleSource(GetItemSource(self.source, name)),
+                    **options,
+                )
+            return result
```

Comment:
`unpack_var_sequence` is supposed to return a list not a dict.  I think this just happens to work because if you iterate over a dict you get its keys.